File: Muc_Do_3_Kha_Kho/Bai11_TrungBinhCong/Solution.cpp
#include <bits/stdc++.h>
using namespace std;
// Bai 11: Trung binh cong >= K
long long bit_arr[200005];
void update(int idx, int val, int n) {
    for (; idx <= n; idx += idx & -idx) bit_arr[idx] += val;
}
long long query(int idx) {
    long long sum = 0;
    for (; idx > 0; idx -= idx & -idx) sum += bit_arr[idx];
    return sum;
}
int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    if (fopen("BAI11.INP", "r")) {
        freopen("BAI11.INP", "r", stdin);
        freopen("BAI11.OUT", "w", stdout);
    }
    int n;
    long long k;
    if (!(cin >> n >> k)) return 0;
    vector<long long> a(n + 1, 0);
    vector<long long> pref(n + 1, 0);
    vector<long long> vals;
    vals.push_back(0);
    for (int i = 1; i <= n; ++i) {
        cin >> a[i];
        a[i] -= k;
        pref[i] = pref[i-1] + a[i];
        vals.push_back(pref[i]);
    }
    sort(vals.begin(), vals.end());
    vals.erase(unique(vals.begin(), vals.end()), vals.end());
    
    long long ans = 0;
    int sz = vals.size();
    for (int i = 0; i <= n; ++i) {
        int pos = lower_bound(vals.begin(), vals.end(), pref[i]) - vals.begin() + 1;
        ans += query(pos);
        update(pos, 1, sz);
    }
    cout << ans << "\n";
    return 0;
}

File: Muc_Do_3_Kha_Kho/Bai11_TrungBinhCong/Solution.py
import sys
def solve():
    try:
        sys.stdin = open('BAI11.INP', 'r')
        sys.stdout = open('BAI11.OUT', 'w')
    except:
        pass
    data = sys.stdin.read().split()
    if not data: return
    n = int(data[0])
    k = int(data[1])
    arr = [int(x) - k for x in data[2:2+n]]
    pref = [0] * (n + 1)
    for i in range(n):
        pref[i+1] = pref[i] + arr[i]
    
    vals = sorted(list(set(pref)))
    def bisect_lower(arr, val):
        l, r = 0, len(arr) - 1
        while l <= r:
            m = (l + r) // 2
            if arr[m] < val: l = m + 1
            else: r = m - 1
        return l
    
    bit = [0] * (len(vals) + 1)
    def update(idx, val):
        while idx < len(bit):
            bit[idx] += val
            idx += idx & (-idx)
            
    def query(idx):
        s = 0
        while idx > 0:
            s += bit[idx]
            idx -= idx & (-idx)
        return s
        
    ans = 0
    for i in range(n + 1):
        pos = bisect_lower(vals, pref[i]) + 1
        ans += query(pos)
        update(pos, 1)
    print(ans)
solve()
